ruby-challenges.rb

```diff
@@ -19,17 +28,56 @@
 nums_array2 = [25, 17, 47, 11]
 # Expected output: 100
 
+def sum(array)
+  array.sum
+end
 
+p sum nums_array1
+p sum nums_array2
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
-
-
 # -------------------3b) Add the ability to pedal faster and brake. The pedal_faster method should increase the speed by a given amount. The brake method should decrease the speed by a given amount. The bike cannot go negative speeds.
 
 # Expected output example: my_bike.pedal_faster(10) => 10
 # Expected output example: my_bike.pedal_faster(18) => 28
 # Expected output example: my_bike.brake(5) => 23
 # Expected output example: my_bike.brake(25) => 0
+
+class Bike
+  attr_accessor :model, :wheels, :current_speed
+  def initialize(model, wheels = 2, current_speed = 0)
+    @model = model
+    @wheels = wheels
+    @current_speed = current_speed
+  end
+
+  def bike_info
+    "The #{model} bike has #{wheels} wheels and is going #{current_speed} mph"
+  end
+
+  def faster(num)
+    @current_speed += num
+  end
+
+  def slower(num)
+    @current_speed -= num
+    if @current_speed <= 0
+      @current_speed = 0
+      p 'You are stopped.'
```

Comment:
Due to Ruby's implicit return you can just have the `p` on the method call and get the string on line 69 as a return if the condition described on line 67 is met.